### PR TITLE
fix: [CDS-53362]: fixed to clear the child(dependent field) dropdown values on parent field changes

### PR DESCRIPTION
**Pull request by sahithikolichala**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AzureArtifacts/AzureArtifacts.tsx [hunk 1]
```diff
@@ -307,6 +328,7 @@ function FormComponent(
                       ...(isValueFixed(formik.values?.package) && { package: '' }),
                       ...(isValueFixed(formik.values?.version) && { version: '' })
                     })
+                    setFeedsData([])
```
Comment (pankajsingh-harness) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AzureArtifacts/AzureArtifacts.tsx:331]: Here we have to reset all the tree fields data. So basically we have to set those options whose values we are updating.

Comment (pankajsingh-harness) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AzureArtifacts/AzureArtifacts.tsx:331]:
```suggestion
                    setFeedsData([])
                    setPackageData([])
                    setVersionData([])
```

**Review by pankajsingh-harness — CHANGES_REQUESTED** [review]

**Comment by sahithikolichala** [comment]
retrigger cypress-rest

**Comment by sahithikolichala** [comment]
retrigger cypress-pipeline

**Comment by sahithikolichala** [comment]
retrigger titlecheck

**Comment by sahithikolichala** [comment]
retrigger titlecheck

**Comment by sahithikolichala** [comment]
retrigger cypress-cv

**Comment by sahithikolichala** [comment]
retrigger jest

**Comment by sahithikolichala** [comment]
retrigger coverage

**Review by cjlee01 — APPROVED** [review]

**Review by pankajsingh-harness — APPROVED** [review]

**Pull request merged**